sdk/management/docs/AUTH.md

```diff
@@ -0,0 +1,100 @@
+# Authentication in Azure Management Library for Java
+
+To use the APIs in the Azure Management Library for Java, as the first step you need to create an authenticated client. This document is to introduce several possible approaches for authentication.
+
+## Prerequisites
+
+Before authenticating the Azure client, there are a few authentication classes you need to know before you decide which kind of authentication would meet your requirement.
+
+* TokenCredential
+  * The `TokenCredential` is an interface in the `azure-core` package for credentials that can provide a token. 
+  * Azure Identity offers multiple implementations of the `TokenCredential` class in the `azure-identity` package. To learn more, see [credentials in Azure Identity](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/identity/azure-identity#credentials).
+
+Sample code to create a simple `ClientSecretCredential`:
+
+```java
+ClientSecretCredential clientSecretCredential = new ClientSecretCredentialBuilder()
+    .clientId("<YOUR_CLIENT_ID>")
+    .clientSecret("<YOUR_CLIENT_SECRET>")
+    .tenantId("<YOUR_TENANT_ID>")
+	.authorityHost("<AZURE_AUTHORITY_HOST>")
+    .build();
+```
+
+Please note, most of credentials require a [service principal registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal). Alternatively, you could also register a service principal in [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli) or [Azure Powershell](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-authenticate-service-principal-powershell).
+
+* AzureProfile
+  * The `AzureProfile` is a class holding `AzureEnvironment`, `subscriptionId`, `tenantId` to configure the requests sending to wire. 
+  * The `subscriptionId` is mandantory for most resource management while the `tenantId` would be required only for Graph RBAC. They can be set via environment variables.
```

Comment:
This could link to #Authenticated part, which help users know they can fetch these two id during authentication. Meanwhile, I think the `null subsId` exception should also point it out.